File: libs/sdk-dotnet/Models/Session.cs
// Copyright Daytona Platforms Inc.
// SPDX-License-Identifier: Apache-2.0

using System.Text.Json.Serialization;

namespace Daytona.Sdk.Models;

public record Session
{
    [JsonPropertyName("sessionId")] public string SessionId { get; init; } = string.Empty;
    [JsonPropertyName("commands")] public List<Command> Commands { get; init; } = new();
}

File: libs/sdk-java/src/main/java/io/daytona/sdk/model/Session.java
// Copyright Daytona Platforms Inc.
// SPDX-License-Identifier: Apache-2.0

package io.daytona.sdk.model;

public class Session extends io.daytona.toolbox.client.model.Session {
    public Session() {}

    public Session(io.daytona.toolbox.client.model.Session source) {
        super();
        if (source != null) {
            setSessionId(source.getSessionId());
            setCommands(source.getCommands());
        }
    }
}
